fw/t/unit/test_http_parser_common.h

```diff
@@ -0,0 +1,699 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+#ifndef __TFW_HTTP_PARSER_COMMON_H__
+#define __TFW_HTTP_PARSER_COMMON_H__
+
+/*
+ * Need to define DEBUG before first the inclusions of
+ * lib/log.h and linux/printk.h.
+ */
+#if DBG_HTTP_PARSER > 0
+#undef DEBUG
+#define DEBUG DBG_HTTP_PARSER
+#endif
+
+#include <linux/types.h>
+#include <asm/fpu/api.h>
+#include <linux/vmalloc.h>
+
+#include "test.h"
+#include "helpers.h"
+#include "fuzzer.h"
+
+#ifdef EXPORT_SYMBOL
+#undef EXPORT_SYMBOL
+#endif
+#define EXPORT_SYMBOL(...)
+
+#include "http_parser.h"
+#include "http_sess.h"
+#include "str.h"
+#include "ss_skb.h"
+#include "msg.h"
+#include "http_msg.h"
+
+static const unsigned int CHUNK_SIZES[] = { 1, 2, 3, 4, 8, 16, 32, 64, 128,
+                                   256, 1500, 9216, 1024*1024
+                                  /* to fit a message of 'any' size */
+                                 };
+static unsigned int chunk_size_index = 0;
+#define CHUNK_SIZE_CNT ARRAY_SIZE(CHUNK_SIZES)
+
+enum {
+	CHUNK_OFF,
+	CHUNK_ON
+};
+
+#define TOKEN_ALPHABET		"!#$%&'*+-.0123456789ABCDEFGHIJKLMNOPQ"	\
+				"RSTUVWXYZ^_`abcdefghijklmnopqrstuvwxyz|~"
+#define QETOKEN_ALPHABET	TOKEN_ALPHABET "\"="
+#define OTHER_DELIMETERS	"(),/:;<=>?@[\\]{}"
+#define OBS_TEXT		"\x80\x90\xC8\xAE\xFE\xFF"
+#define ETAG_ALPHABET		OTHER_DELIMETERS TOKEN_ALPHABET OBS_TEXT
+#define VCHAR_ALPHABET		"\x09 \"" OTHER_DELIMETERS 		\
+				TOKEN_ALPHABET OBS_TEXT
+
+#define EXPECT_BLOCK_DIGITS(head, tail, BLOCK_MACRO)		\
+	BLOCK_MACRO(head tail);					\
+	BLOCK_MACRO(head "  " tail);				\
+	BLOCK_MACRO(head "5a" tail);				\
+	BLOCK_MACRO(head "\"" tail);				\
+	BLOCK_MACRO(head "=" tail);				\
+	BLOCK_MACRO(head "-1" tail);				\
+	BLOCK_MACRO(head "0.99" tail);				\
+	BLOCK_MACRO(head "dummy" tail);				\
+	BLOCK_MACRO(head "4294967296" tail);			\
+	BLOCK_MACRO(head "9223372036854775807" tail);		\
+	BLOCK_MACRO(head "9223372036854775808" tail);		\
+	BLOCK_MACRO(head "18446744073709551615" tail);		\
+	BLOCK_MACRO(head "18446744073709551616" tail)
+
+#define EXPECT_BLOCK_SHORT(head, tail, BLOCK_MACRO)		\
+	BLOCK_MACRO(head "65536" tail);				\
+	BLOCK_MACRO(head "2147483647" tail);			\
+	BLOCK_MACRO(head "2147483648" tail);			\
+	BLOCK_MACRO(head "4294967295" tail)
+
+/* For ETag and If-None-Match headers */
+#define COMMON_ETAG_BLOCK(head, BLOCK_MACRO)			\
+	BLOCK_MACRO(head "\"dummy");				\
+	BLOCK_MACRO(head "dummy\"");				\
+	BLOCK_MACRO(head "'dummy'");				\
+	BLOCK_MACRO(head "W/ \"dummy\"");			\
+	BLOCK_MACRO(head "w/\"dummy\"");			\
+	BLOCK_MACRO(head "\"\x00\"");				\
+	BLOCK_MACRO(head "\"\x0F\"");				\
+	BLOCK_MACRO(head "\"\x7F\"");				\
+	BLOCK_MACRO(head "\" \"");				\
+	BLOCK_MACRO(head "\"\"\"")
+
+
```

Comment:
There are many places of multiple empty lines - we use not more that 1 empty line in C/C++ code, so could you please adjust empty lines.